src/otx/algo/detection/ssd.py

```diff
@@ -53,6 +64,94 @@ def device(self) -> device:
         self.classification_layers = self.get_classification_layers(self.config, MODELS, "model.")
         return build_mm_model(self.config, MODELS, self.load_from)
 
+    def setup_callback(self, trainer: Trainer) -> None:
```

Comment:
I was wondering if `SSD` model has access to `Trainer`? or a callback? Thought it is just a torch model, no?

Reply:
![image](https://github.com/openvinotoolkit/training_extensions/assets/106788349/b1e7cfb5-9518-40ec-80dc-b22c662152fa)
Here, OTXLitModule transfer trainers to OTXSSD

Reply:
Thanks for the clarification @jaegukhyun.
This may be not a comment to this PR only, but a comment in general. 

In my mind, I was thinking that OTXModel is Lightning free that only has a forward-pass mechanism with pre and post processing capabilities. Adding Lightning-related stuff to OTXModel might cause coupling across the library. Especially as the library grows. When these two objects shares stuff bidirectionally, then it might raise a question of why there are two objects exists, but not one? Does it make sense?

Reply:
I think your statement is make sense. @vinnamkim What do you think of this idea? I'm not sure about pros and cons between callback and direct linking

Reply:
As same as this comment https://github.com/openvinotoolkit/training_extensions/pull/2902#discussion_r1489655719, I'm thinking about refactoring to deprecate `OTXLitModule` and promote `OTXModel` to `LightningModule`. I'll make a schedule for this work after discussion with Wonju and Mark.

Reply:
Yep, then it makes sense